realm-annotations-processor/src/main/java/io/realm/processor/RealmProcessor.java

```diff
@@ -30,15 +31,73 @@
 
 import io.realm.annotations.RealmClass;
 
+/**
+ * The RealmProcessor is responsible for creating the plumbing that connects the RealmObjects to a Realm. The process
+ * for doing so is summarized below and then described in more detail.
+ *
+ *
+ * DESIGN GOALS
+ *
+ * The processor should support the following design goals:
+ *
+ * - Minimize reflection.
+ * - As much as possible can be obfuscated.
```

Comment:
"It can be obfuscated as much as possible"
Otherwise I read it with Yoda's voice in my head :)